documentation/architecture/ADR1:dependency-injection.md

```diff
@@ -0,0 +1,76 @@
+## Context
```

Comment:
nit: You should probably make the directory sort by default. Maybe: `adr-0001-dependency-injection.md`

I don't like the colon: it's an atypical character in file names and it may need special handling in some applications.